java/src/test/java/com/kobiton/usecases/practices/RetrieveAutomationDetails.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Created on Sep 26, 2017, 5:17:43 PM
+ *
+ * Copyright(c) 2017 Kobiton Inc.  All Rights Reserved.
+ * This software is the proprietary information of Kobiton Company.
+ *
+ */
+package com.kobiton.usecases.practices;
+
+import com.kobiton.usecases.*;
+import io.appium.java_client.android.AndroidDriver;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.concurrent.TimeUnit;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class RetrieveAutomationDetails extends BaseTest {
+    
+    @Test()
+    public void testGetAllSessions() throws IOException {
+        //API details: https://api.kobiton.com/docs/?java#get-sessions
+        String url = String.format("https://%s/v1/sessions", getHostName());
+        
+        OkHttpClient client = new OkHttpClient();
+        Request request = new Request.Builder()
+                .url(url)
+                .header("Authorization", getAuthorization())
+                .build();
+        
+        Response response = client.newCall(request).execute();
+        
+        Assert.assertNotNull(response.body(), "Response body is null");
+        
+        // responseBodyString is JSON format, you could parse to use it as your demand
+        String responseBodyString = response.body().string();
+        Assert.assertNotNull(responseBodyString, "Response body string is null");
+    }
+    
+    @Test
+    public void testGetSessionDetailAfterExecuteTest() throws MalformedURLException, IOException {
+        AndroidDriver<WebElement> driver = executeSimpleAppTest();
+        
+        String kobitonSessionId = driver.getSessionDetails().get("kobitonSessionId").toString();
+        
+        String url = String.format("https://%s/v1/sessions/%s", getHostName(), kobitonSessionId);
+        
+        OkHttpClient client = new OkHttpClient();
+        Request request = new Request.Builder()
+                .url(url)
+                .header("Authorization", getAuthorization())
+                .build();
+        
+        Response response = client.newCall(request).execute();
+        Assert.assertNotNull(response.body(), "Response body is null");
+        
+        // responseBodyString is JSON format, you could parse to use it as your demand
+        String responseBodyString = response.body().string();
+        System.err.println("response body: " + responseBodyString);
+        Assert.assertNotNull(responseBodyString, "Response body string is null");
+    }
+
+    private AndroidDriver<WebElement> executeSimpleAppTest() throws MalformedURLException {
```

Comment:
why do we have a test here?

Reply:
In this case, we execute a automation test, then use api to get detailed result.